### ISSUE TITLE
Using discriminator as loss?

### ISSUE THREAD

**Issue by markemus** [eladrich/pixel2style2pixel#106]
I saw this line in the paper:

"Besides the simplification of the training process, as no adversary discriminator needs to be trained, using a pretrained StyleGAN generator offers several intriguing advantages over previous works"

Did you guys ever try using a discriminator instead of your three loss functions? I'm trying to train on a different modality (tissue samples) and I'm struggling to replace the face-specific losses. L2 alone doesn't work too well.

**Comment by yuval-alaluf** [comment]
We actually tried using the discriminator as an added loss, but did not find much improvement in the result. I would recommend trying to incorporate both the L2 and LPIPS losses into your training. 
I also believe that incorporating a loss function that explicitly handles the preservation of fine details will provide the most improvement in your case.

**Comment by markemus** [comment]
@yuval-alaluf thanks! I tried with both L2 and LPIPs and just L2- any suggestions for hyperparams to play with?

**Comment by yuval-alaluf** [comment]
I would also recommend you take a look at the MoCo-based similarity loss used in the recent e4e encoder. Here's the link to the repo: 
https://github.com/omertov/encoder4editing
Basically, in the work, they replace the ID loss, which is specific to human faces with a loss computed using the embedding from a MoCo-based ResNet trained on ImageNet. Since the features are more general, the MoCo loss is more suitable for non-facial domains. 
I am a co-author of the e4e paper so feel free to ask any questions here or in the linked repo. @eladrich and I also plan on adding support for the loss in this pSp repo in the coming weeks.

**Comment by markemus** [comment]
Thanks yuval! I'm testing out that project now- looking forward to the update.
